src/sleep/sleep.c

```diff
@@ -62,15 +62,31 @@ static int write_hibernate_location_info(void) {
                                        "Invalid hibernate type: %s", type);
 
         /* Only available in 4.17+ */
-        if (access("/sys/power/resume_offset", W_OK) < 0) {
+        if (access("/sys/power/resume_offset", R_OK) < 0) {
                 if (errno == ENOENT) {
                         log_debug("Kernel too old, can't configure resume offset, ignoring.");
                         return 0;
                 }
 
-                return log_debug_errno(errno, "/sys/power/resume_offset not writeable: %m");
+                return log_debug_errno(errno, "/sys/power/resume_offset not readable: %m");
         }
 
+        /* use resume_offset set by user if available */
+        r = read_one_line_file("/sys/power/resume_offset", &resume_offset);
+        if (r < 0)
+                log_warning_errno(r, "Error reading from /sys/power/resume_offset: %m");
+
+        r = safe_atou64(resume_offset, &offset);
```

Comment:
please, let's not try to parse `resume_offset` if it's not initialized anyway since read_one_line_file() failed, while this will just fail, it's still ugly, since this means we create two error messages here, and it really looks like a real bug that way